### PR TITLE
Update coiled_function decorator

### PR DESCRIPTION
**Pull request by mrocklin**
Two changes:

1.  Remove the old functools.wraps functionality, now that upstream handles this for us.
2.  Add the module name into the cluster name, so that we can run duckdb and polars simultaneously more easily

### CODE REVIEW

**Comment by mrocklin** [comment]
Still working on this.

File: tests/tpch/conftest.py [hunk 1]
```diff
@@ -134,14 +135,17 @@ def spark(spark_setup, benchmark_time):
 
 def coiled_function(**kwargs):
     def _(function):
+        frame = sys._current_frames()[threading.get_ident()]
+        filename = frame.f_back.f_code.co_filename
+        module = filename.split("_")[-1].split(".")[0]  # dask, spark, ...
```
Comment (jrbourbeau) [inline tests/tpch/conftest.py:140]: Just a driveby comment but maybe `inspect.getmodule` or `function.__module__` is useful here

Comment (mrocklin) [reply tests/tpch/conftest.py:140]: Thanks!  That's much nicer.

**Comment by mrocklin** [comment]
OK, this is feeling good now.

**Review by milesgranger — APPROVED** [review]
A go from me, assuming tests are happy. :+1:

**Comment by mrocklin** [comment]
Test failure points to this run: https://cloud.coiled.io/clusters/294695/information?account=dask-benchmarks&computation=d9e9bd9d-4ef9-4455-8ff9-7c8206bb4344&sinceMs=1697656440506&untilMs=1697656440764&tab=Code&filterPattern=

I'm not getting any information from the platform, even logs.  Things just hang.  @dchudz I'm curious if you all are able to see anything more deeply here on this run.

**Comment by dchudz** [comment]
@mrocklin the cluster has plenty of logs.

is your link intentionally only for a specific time range?

my interpretation would be that nothing happened in that time range. 

I can't say more without knowing more about how the benchmarks work or how that link was put together.

**Comment by dchudz** [comment]
oh, maybe that's a link for an *empty* time range?

Screen Shot 2023-10-18 at 4 26 10 PM

**Comment by mrocklin** [comment]
Oh odd.  I clicked on the code block for `test_query_2`, the query that failed, and that fits in that 20 minute block of emptiness (I think this is also the timeout period with pytest) assuming that it was taking up that 20 minute block.  It looks like I was mistaken though.  

cc'ing also @fjetter in case he can help to understand it.  Maybe flakiness on Dask's part?

**Comment by dchudz** [comment]
probably this isn't very interesting, but:

* Matt's original link is a zero-minute block
* in the 20-minute block of emptiness ([new link](https://cloud.coiled.io/clusters/294695/information?account=dask-benchmarks&computation=&tab=Metrics&filterPattern=&sinceMs=1697656446977&untilMs=1697657636977)) there are metrics, but indeed no tasks, and no logs
